clang/test/CIR/func-simple.cpp

```diff
@@ -0,0 +1,53 @@
+// Simple functions
+// RUN: %clang_cc1 -std=c++20 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir %s -o -  | FileCheck %s
+
+void empty() { }
+// CHECK: cir.func @empty() -> !cir.void {
```

Comment:
Why aren't the function names mangled?

Reply:
The names aren't mangled because the name mangling code hasn't been upstreamed yet.  (Name mangling is part of the ABI which is part of CodeGen which means the existing Clang code for that can't be used and ClangIR needs its own copy of the ABI code even though there is nothing ClangIR-specific about it.  At least that's how I think it works.)

Reply:
Ideally once there will be abi-lib: https://discourse.llvm.org/t/llvm-abi-a-library-for-generating-abi-compliant-llvm-ir/37196/11 to be used across codegens.